### ISSUE TITLE
High Level and Low Level API's

### ISSUE THREAD

**Issue by yisraelU** [svroonland/zio-amqp#201]
Current design exposes connection and channel , which is not bad. I am thinking it might be valuable to have a High level api which i represented as a Consumer or a Publisher (or both) which would hide the channel and connection and just expose the consuming/publishing methods on it .
Thinking this would also  be a good place to implement a connection retry plus auto reconnect feature
@svroonland 
Thoughts ?

**Comment by svroonland** [comment]
Yes I like that a lot. Was thinking about it a while back. The channel per thread design that rabbitmq  does not necessarily make sense within a ZIO (fiber) environment and where we have better control over concurrency.

If you have an API design in mind, we could start with that.
